deployment/src/test/java/io/quarkiverse/cxf/deployment/test/CxfClientTest.java

```diff
@@ -16,15 +21,50 @@
     public static final QuarkusUnitTest test = new QuarkusUnitTest()
             .setArchiveProducer(() -> ShrinkWrap.create(JavaArchive.class)
                     .addClass(FruitWebService.class)
+                    .addClass(FruitWebServiceImpl.class)
                     .addClass(Fruit.class))
             .withConfigurationResource("application-cxf-test.properties");
 
+    @Inject
+    @Named("io.quarkiverse.cxf.deployment.test.FruitWebService")
+    CXFClientInfo clientInfo;
+
     @Inject
     FruitWebService clientService;
 
+    @Inject
+    @CXFClient
+    FruitWebService clientProxy;
```

Comment:
What is the difference between cliebtproxy and clientservice? So why add not needed annotation?

Reply:
Hmm I see that one is proxy but what is goal behind switch to proxy

Reply:
This is the difference, addressing issue #182:

```
    @Inject
    @CXFClient
    FruitWebService clientProxy;
```

Here @CXFClient asks to insert unconditionally a CXF proxy client. When you use this annotation, you know for sure that you are talking via a proxy to your backend service. You can also configure your "client" for logging and so on.

Consider the situation before:

```
    @Inject
    SomeWebService client;        // SomeWebService supposed to be a SEI
```

What is getting injected here? You need to investige SomeWebService to know. If there is a impl of SEI in your classpath, then you get no proxy at all - you just get the impl itself injected. Contrary, if no impl is around, you get a proxy injected:

```
    @WebService
    public class SomeWebServiceImpl implements SomeWebService {}      // injects  SomeWebServiceImpl

    vs

    @WebService
    public interface SomeWebService {}                                 // injects Proxy (via Producer).
``

This behaviour is rather confusing im my opinion (see #182).

The new annotation @CXFClient resolves this issue. Once attached, you indicate that you want to have a proxy injected. In addition, you can indicate which particular client configuration you want to have.

Contrary to this you may want to have the service implementation injected:

```
    @Inject
     SomeWebService client;    // CDI error if NO implementation is around.
```